### DISCUSSION TITLE
Order of element numbers

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#25226] [Q&A Meshing] [answered]
All,
I am adding more elements into my model. The element numbers are jumping now; for example, 1,2, ... 99, 100, and then the number jumps to 201, 202, ... 299, 300, and then the number jumps to 401, 402, ... 499, 500, and so on.

I am wondering if the element numbers jump like this, it causes a problem(s) of the time step size because it converges but the time step size is so small.
Please help.
S. Thomas

**Comment by GiudGiud** [comment]
Hello

The element ids jumping is not an issue for the solver.
You can disable element renumbering if this is a renumbering issue in the `[Mesh]` block

How low does the time step size end? What kind of physics is this?

Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.
The very simplified model and transient executioner we are using are shown in the attached sketch.
The sketch shows to solve two diffusion equations for two blocks with flux applied around the interfaces of blocks but the actual model has numerous blocks and the actual ratio of the vertical direction dimension to the horizontal direction dimension is ~200 : 1. Also, the block numbers are jumping: 1 ... 60, 100 … 165, 200 … 145, and so on.
It converges but the time step size goes to as small as 0.001 depending on the selection of  l_tol, nl_abs_tol, nl_max_its, and so on, of course. 
Please let me know if you have any suggestions to make it better.
I would like to thank you for your time and help.
S. Thomas Yokuda
[Interfaces_two_blocks.pdf](https://github.com/idaholab/moose/files/12441760/Interfaces_two_blocks.pdf)

**Reply by GiudGiud** [reply]
A few changes I would make:
- turn on verbose in the executioner and look at the scaling factors. 
If they are forcing too tight a convergence by being recomputed, I d remove the compute_scaling_once = false 
- linear tolerance is too tight
- remove the l2 line_search, specify `none` instead. Line searches have never really proven that useful, we need to re-visit them

**Reply by styyokuda** [reply]
Thank you, Guillaume.
The automatic scaling factors for some variables go up to 9e+08. Is this OK?
Please help.
S. Thomas

**Reply by GiudGiud** [reply]
No this seems very high. I would have "compute_scaling_once = true", the default, to avoid this.

**Reply by styyokuda** [reply]
Thank you, Guillaume.

As the time develops, the time step size keeps getting smaller.  Does it indicate anything?
I don't have a lot of experience with unsteady modeling.  So, please help.
S. Thomas

**Reply by GiudGiud** [reply]
problem is getting harder to solve. 
This might be because the convergence criteria is getting hard to meet. If things are converging to a steady realm, the initial residual is going down so a relative convergence criteria is getting harder and harder to meet

Can you attach the log please?

**Reply by styyokuda** [reply]
Thank you, Guillaume.
The model has source terms as a function of time. So, it does not have a steady solution.
Is the "log" you mean the output printout on the monitor?
Please let me know.
S. Thomas

**Reply by GiudGiud** [reply]
yes to your second question.
Does the change slow down though?

**Reply by styyokuda** [reply]
Thank you, Guillaume.
So, it seems that a finer mesh is needed but a finer mesh is not possible for the entire global computational domain.  I would like to ask if a finer mesh is used only for some blocks (e.g., Block 2 for the attached sketch) and a coarse mesh is used for the rest of the global domain using the MOOSE Apps.  In this case, can different time step sizes be used for the global computational domain and the computational domains of blocks?
Please help? 
S. Thomas
[Interfaces_two_blocks.pdf](https://github.com/idaholab/moose/files/12553915/Interfaces_two_blocks.pdf)

**Reply by GiudGiud** [reply]
Hello

For the spatial mesh:
We can use local refinement, either a mesh construction time using a RefineBlockGenerator or during the simulation using Adaptivity / AMR

For the time grid / steps:
We currently do not have the capability to use different time steps in different subdomains.
We would be very interested to have that as well, we have began to build (and are continuing next year) some capacities towards that. It will not be ready for maybe 1-2 years at least

Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.
The printout on the monitor is too long and hard to see.
Is there any way to suppress the printout on the monitor except for the time information (e.g., time, time steps, and so on) 
Please help. S. Thomas

**Reply by GiudGiud** [reply]
Hello

Please see the console output documentation.
https://mooseframework.inl.gov/source/outputs/Console.html
You can 'hide' a number of postprocessors for example. You can remove linear/nonlinear iteration results. There's quite a bit that can be adjusted


Guillaume

**Reply by styyokuda** [reply]
Thank you, Guillaume.
Attached please find two pdf files “model_console” and “Simplified_Model.”  In “model_console,” I listed the Mesh, Function, Executioner, and Executioner blocks used to run the model and the printout on the console of the model run where 12 processors were used but I deleted the printout from 11 processors to make it look clearer.

The model has seven variables to solve seven diffusion equations.  The challenge is the geometry as the aspect ratio of the horizontal direction dimension to the vertical direction dimension is ~ 1 : 200 and the boundaries of some elements are interfaces to that flux is applied.

The sketch of “Simplified_Model” shows to solve three diffusion equations for three blocks with flux applied around the interfaces of one block and one block of the source to show the concept of our model.

For the run, Variable 6 of Block 6 is not activated until 3600 (seconds) that is Time Step 7 in “model_console.” As you see, the time step size is getting smaller and smaller (1e-12 at Time Step 16).
Please let me know if you see anything to make it better.
Our clients want to make it happen. Please help us.
S. Thomas

[model_console.pdf](https://github.com/idaholab/moose/files/12665334/model_console.pdf)
[Simplified_Model.pdf](https://github.com/idaholab/moose/files/12665336/Simplified_Model.pdf)

**Reply by GiudGiud** [reply]
Hello

We ll need more outputs than this. Step 7 does not converge, why?
Can you print the linear and nonlinear residuals? in the outputs block

Can you paste the rest of the input for the equations? I would want to check that things are properly block-restricted

Guillaume

**Reply by styyokuda** [reply]
Dear Guillaume,

It seems that the element ids jumping is causing an issue for the solver (please see our conversation above).
You have suggested "You can disable element renumbering if this is a renumbering issue in the [Mesh] block."
May I ask what "disable element renumbering" means in your suggestion. 
Please help.
S. Thomas

**Reply by GiudGiud** [reply]
It s a parameter called something like 'allow_element_renumbering' to the mesh that you set to false

**Reply by styyokuda** [reply, marked as answer]
Dear Guillaume,
I found that the element ids jumping is "not" causing an issue for the solver.
Thank you very much for your time and great help.
S. Thomas

**Reply by GiudGiud** [reply]
ah that's good. It s not expected to.
